### ISSUE TITLE
Help Wanted 🙏 to run the ray.train._internal.session module

### ISSUE THREAD

**Issue by Mouly22** [ultralytics/ultralytics#10011]
Hello YOLO enthusiasts,
It will be great if anyone can help me to solve the below shown issue:

Here is the link of my notebook: https://www.kaggle.com/code/ikramulkayes/yolov5-test1/notebook
The model was working fine around 3 weeks ago.  However, for some unknown reason, now it is showing error. 
We have also tried to use large model. Please let me know how can we solve this problem. 
`from ultralytics import YOLO
model = YOLO("yolov5m.pt")
model.train(data="/kaggle/working/yolov5.yaml", epochs=3)`

![image](https://github.com/ultralytics/ultralytics/assets/95493187/50b4aab4-e4f8-420a-a01c-070cc66f0453)

**Comment by glenn-jocher** [comment]
Hello! Thank you for reaching out for help with your YOLOv8 model issue. Based on the code snippet you've shared, it looks like there's a slight mix-up in the model version and the import statement you're using.

Please make sure you're using the correct version of the Ultralytics library for YOLOv8 models. If your implementation is aimed at YOLOv8, you should replace `"yolov5m.pt"` with a corresponding YOLOv8 model file, like `"yolov8n.pt"` (note the change from v5 to v8 in the model name).

Here is the updated code snippet:
```python
from ultralytics import YOLO
model = YOLO("yolov8n.pt") 
model.train(data="/kaggle/working/yolov5.yaml", epochs=3)
```

Please also ensure that the `.yaml` file is compatible with YOLOv8. The filename suggests it might be configured for YOLOv5. You might need to adjust it to match the requirements of YOLOv8's architecture and data handling.

Regarding the `ray.train._internal.session module` mention in your comment, it seems like there might be some confusion as the provided code or issue does not explicitly reference Ray's training session module. If your project involves distributed training using Ray and you're encountering specific errors related to it, could you please provide more details or error messages related to `ray.train._internal.session`?

For any further assistance, feel free to share additional details or error logs. We're here to help! 🚀

**Comment by Dimasilham7** [comment]
same here, code looks fine before, but in the recent shows same error, already tried to install and force install ray[tune] but cant help at all

**Comment by Dimasilham7** [comment]
> Hey there! 🌟 Sorry to hear you're hitting a snag with the recent error. If you've already tried reinstalling ray[tune] without success, you might want to ensure your environment is up to date. Here's a quick snippet you could run to update `ray` and `tune`:
> 
> ```shell
> pip install -U ray[tune]
> ```
> 
> If the issue persists, sharing a bit more about the error message or code snippet might help us pinpoint the problem. Keep us posted! 🚀

Hey glenn, already implemented what you tell me to do, i think this problem is caused from kaggle since i had same setting and library in the locas, but weirdly seems fine on local but when i run it on kaggle it doesn't

**Comment by glenn-jocher** [comment]
Hey there! Thanks for the update. It's interesting that the issue surfaces on Kaggle but not locally. It might be due to environment discrepancies between Kaggle and your local setup. As a quick check, could you ensure that the version of `ray` and other dependencies on Kaggle is the same as on your local machine? You can check the versions using:

```shell
pip list | grep ray
```

Another workaround you might try is using a Docker container for Kaggle that mirrors your local environment more closely. Sometimes aligning environments can help isolate these quirky behaviors.

Let's keep troubleshooting this! 🛠️

**Comment by Dimasilham7** [comment]
> Hey there! Thanks for the update. It's interesting that the issue surfaces on Kaggle but not locally. It might be due to environment discrepancies between Kaggle and your local setup. As a quick check, could you ensure that the version of `ray` and other dependencies on Kaggle is the same as on your local machine? You can check the versions using:
> 
> ```shell
> pip list | grep ray
> ```
> 
> Another workaround you might try is using a Docker container for Kaggle that mirrors your local environment more closely. Sometimes aligning environments can help isolate these quirky behaviors.
> 
> Let's keep troubleshooting this! 🛠️

yes, I've trying to workaround on this, this is my pip show ray on local:
C:\Users\user>pip show ray
WARNING: Package(s) not found: ray

showing i didn't use ray, and this is my ray on kaggle:

WARNING: No metadata found in /opt/conda/lib/python3.10/site-packages
WARNING: No metadata found in /opt/conda/lib/python3.10/site-packages
WARNING: No metadata found in /opt/conda/lib/python3.10/site-packages
WARNING: No metadata found in /opt/conda/lib/python3.10/site-packages
WARNING: No metadata found in /opt/conda/lib/python3.10/site-packages
WARNING: No metadata found in /opt/conda/lib/python3.10/site-packages
WARNING: No metadata found in /opt/conda/lib/python3.10/site-packages
array-record                             0.4.0
ray                                      2.20.0
ray-cpp                                  2.5.1
xarray                                   2023.8.0
xarray-einstats                          0.6.0
Note: you may need to restart the kernel to use updated packages.

after i do installing this package !pip install ray[tune]==2.6.3 ultralytics

**Comment by Dimasilham7** [comment]
hey, i think i just fix this with using this pip install --upgrade ray, silly me not checking it with this before

**Comment by glenn-jocher** [comment]
@Dimasilham7 hey! Glad to hear that upgrading Ray did the trick for you! It's often easy to overlook such updates. If you run into any more snags or have further questions, feel free to reach out. Happy coding! 🚀

**Comment by Anyapara** [comment]
@Dimasilham7 , i did  pip install --upgrade ray and still it didn't work in Kaggle, kindly share how you solved yours

**Comment by ousaghro** [comment]
i solved the problem using pip install --upgrade ray thank you very much

**Comment by glenn-jocher** [comment]
Great to hear you resolved the issue by upgrading Ray! If you have any more questions, feel free to ask.

**Comment by chg0901** [comment]
> @Dimasilham7 , i did pip install --upgrade ray and still it didn't work in Kaggle, kindly share how you solved yours

I also get this problems with torch2.0.0, ultralytics 8.3.58 and ray-2.40.0

**Comment by UltralyticsAssistant** [comment]
Great news! A potential fix for this issue has been merged in [PR #19144](https://github.com/ultralytics/ultralytics/pull/19144). 🎉 

### Key Changes from the PR:
This PR updates the Ray Tune integration to ensure compatibility with the latest Ray versions by replacing deprecated methods:
- Updated the internal Ray Tune session call from `ray.tune.is_session_enabled()` to `ray.train._internal.session.get_session()`.
- Bumped the Ultralytics version from `8.3.73` to `8.3.74`.

These changes aim to reduce runtime errors and improve stability for users incorporating Ray Tune into their workflows.

### Testing the Fix:
To try out the updated code and confirm if it resolves the issue:
1. Install the latest changes directly from the `main` branch:
   ```bash
   pip install git+https://github.com/ultralytics/ultralytics.git@main
   ```
2. Or, wait for the next official PyPI release.

### Thanks and Feedback:
A huge thank you to @Y-T-G for authoring the PR, with valuable contributions from @glenn-jocher and @ppraneth. 🙌 Your efforts are greatly appreciated!

If this fix resolves the issue, it would be wonderful to hear back. On the other hand, if the issue persists, any additional details would help ensure it is fully addressed. Thanks again for reporting this, and happy testing! 🚀
